src/Forms/GridField/GridFieldFilterHeader.php

```diff
@@ -103,6 +109,7 @@ protected function checkDataType($dataList)
         if ($dataList instanceof Filterable) {
             return true;
         } else {
+            // This will be changed to always throw an exception in a future major release.
```

Comment:
Adding this note so it's immediately very clear what the expected end result is when we remove this deprecated code in a future major release. Same with the other classes where the same note is added.